addons/account/models/account_requested_document.py

```diff
@@ -0,0 +1,41 @@
+import base64
+import uuid
+
+from odoo import models, fields, http, _
+
+
+class AccountRequestedDocument(models.Model):
+    _name = 'account.requested.document'
+    _description = 'Accounting Requested Document(A technical model for sharing request document link)'
+
+    name = fields.Char('Document Name')
+    attachment_id = fields.Many2one('ir.attachment', help='Technical field to hold the document')
+    token = fields.Char(required=True, default=lambda __: str(uuid.uuid4()), index=True)
```

Comment:
Token fields should always have a group to avoid portal user reads: 
```suggestion
    token = fields.Char(required=True, default=lambda __: str(uuid.uuid4()), index=True, groups='base.group_user')
```
Your current access rights don't let portal users read the model even without the `groups=` but if they are ever granted the permission, this is a good practice.